src/components/Search/SearchRouter/SearchRouterInput.tsx

```diff
@@ -1,38 +1,71 @@
+import type {StyleProp, ViewStyle} from 'react-native';
+import TextInput from '@components/TextInput';
 import useThemeStyles from '@hooks/useThemeStyles';
 import variables from '@styles/variables';
 import CONST from '@src/CONST';
 
 type SearchRouterInputProps = {
+    /** Callback triggered when the input text changes */
+    onChange?: (searchTerm: string) => void;
+
+    /** Callback invoked when the user submits the input */
+    onSubmit?: () => void;
+
+    /** Whether the input is full width */
     isFullWidth: boolean;
-    onChange: (searchTerm: string) => void;
-    onSubmit: () => void;
+
+    /** Default value for text input */
+    defaultValue?: string;
+
+    /** Whether the input is disabled */
+    disabled?: boolean;
+
+    /** Any additional styles to apply */
+    wrapperStyle?: StyleProp<ViewStyle>;
+
+    /** Any additional styles to apply when input is focused */
+    wrapperFocusedStyle?: StyleProp<ViewStyle>;
+
+    /** Component to be displayed on the right */
+    rightComponent?: ReactNode;
 };
 
-function SearchRouterInput({isFullWidth, onChange, onSubmit}: SearchRouterInputProps) {
+function SearchRouterInput({isFullWidth, onChange, onSubmit, defaultValue = '', disabled = false, wrapperStyle, wrapperFocusedStyle, rightComponent}: SearchRouterInputProps) {
     const styles = useThemeStyles();
 
-    const [value, setValue] = useState('');
+    const [value, setValue] = useState(defaultValue);
+    const [isFocused, setIsFocused] = useState<boolean>(false);
 
     const onChangeText = (text: string) => {
         setValue(text);
-        onChange(text);
+        onChange?.(text);
     };
 
-    const modalWidth = isFullWidth ? styles.w100 : {width: variables.popoverWidth};
+    const inputWidth = isFullWidth ? styles.w100 : {width: variables.popoverWidth};
 
     return (
-        <BaseTextInput
-            value={value}
-            onChangeText={onChangeText}
-            onSubmitEditing={onSubmit}
-            autoFocus
-            textInputContainerStyles={[{borderBottomWidth: 0}, modalWidth]}
-            inputStyle={[styles.searchInputStyle, styles.searchRouterInputStyle, styles.ph2]}
-            role={CONST.ROLE.PRESENTATION}
-            autoCapitalize="none"
-        />
+        <View style={[styles.flexRow, styles.alignItemsCenter, wrapperStyle ?? styles.searchRouterInput, isFocused && wrapperFocusedStyle]}>
+            <View style={styles.flex1}>
+                <TextInput
+                    autoFocus
+                    value={value}
+                    onChangeText={onChangeText}
+                    onSubmitEditing={onSubmit}
+                    role={CONST.ROLE.PRESENTATION}
+                    autoCapitalize="none"
+                    disabled={disabled}
+                    shouldUseDisabledStyles={false}
+                    textInputContainerStyles={[{borderBottomWidth: 0}]}
+                    inputStyle={[styles.searchInputStyle, inputWidth, styles.pl3, styles.pr3]}
```

Comment:
I would prefer leaving it this way, since using `paddingHorizontal` here gets overwritten by internal styling of TextInput, leaving only the right padding intact and nullifying the left one. When specifying the paddings directly they get presented correctly.